aws/vpc/main.tf

```diff
@@ -89,7 +89,28 @@ resource "aws_nat_gateway" "nat_gateway" {
 }
 
 /*
- * Routes for private subnets to use NAT gateway
+ * Set to true create transit gateway attachments in the private subnets and
+ * route traffic to the TGW instead of NAT GW.
+ * Should be used when create_nat_gateway=false.
+ */
+#Create Transit Gateway VPC attachment for orlando-dr to attach to hub.
+resource "aws_ec2_transit_gateway_vpc_attachment" "transit_gateway" {
+  count = var.create_transit_gateway_attachment ? 1 : 0
```

Comment:
If it's important to not create this if `create_nat_gateway` is true, then it might be good to enforce that in the code. For instance:
```
locals (
  create_transit_gateway = var.create_transit_gateway_attachment && !var.create_nat_gateway
)
```
and use `local.create_transit_gateway` in the `count` on this resource and the `aws_route` resource.